webapp/src/components/common/Matrix/components/MatrixUpload/index.tsx

```diff
@@ -0,0 +1,80 @@
+/**
+ * Copyright (c) 2024, RTE (https://www.rte-france.com)
+ *
+ * See AUTHORS.txt
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ *
+ * SPDX-License-Identifier: MPL-2.0
+ *
+ * This file is part of the Antares project.
+ */
+
+/**
+ * MatrixUpload.tsx - Main component that manages matrix uploads
+ */
+import { StudyMetadata } from "@/common/types";
+import DatabaseUploadDialog from "@/components/common/dialogs/DatabaseUploadDialog";
+import FileUploadDialog from "@/components/common/dialogs/FileUploadDialog";
+import { useTranslation } from "react-i18next";
+
+interface MatrixUploadProps {
+  studyId: StudyMetadata["id"];
+  path: string;
+  type: "file" | "database";
+  open: boolean;
+  onClose: VoidFunction;
+  onFileUpload?: (file: File) => Promise<void>;
+  fileOptions?: {
+    accept?: Record<string, string[]>;
+    dropzoneText?: string;
+  };
+}
+
+function MatrixUpload({
```

Comment:
Rename MatrixUploadDialog

Reply:
its only a wrapper for two different types of upload dialogs, not a dialog itself

Reply:
I undestand but it's it's open a Dialog. All dialogs are wrappef or BasicDialog too.

Reply:
Again, the component doesn't implement any dialog functionality itself, it's purely a wrapper that chooses between two different dialog components based on the type prop. Naming it MatrixUploadDialog would imply it's a dialog component, which could mislead other developers about its actual purpose and implementation.